### PR TITLE
Fix: Refactored test_host_node_failure test

### PR DESCRIPTION
**Pull request by nik-redhat**
**Fix:**
Refactored the test test_host_node_failure to check for
MCG health and RGW bucket creation.

Fixes: #5571

Signed-off-by: nik-redhat <nladha@redhat.com>

### CODE REVIEW

**Comment by nik-redhat** [comment]
The first 2 point have not been considered reason being, the test is interlinked between RGW and MCG so I doubt if separating them is possible (required).

The function `wait_for_storage_pods` waits for all the pods to be up, so no separate check for noobaa pods are needed.

File: tests/manage/rgw/test_host_node_failure.py [hunk 1]
```diff
@@ -53,25 +54,35 @@ def init_sanity(self):
         """
         self.sanity_helpers = Sanity()
 
-    def create_obc_creation(self, bucket_factory, mcg_obj, key):
+    def create_obc_creation(self, bucket_factory, interface, obj, key):
```
[the remark this answers is not in the document]
Comment (nik-redhat) [reply tests/manage/rgw/test_host_node_failure.py]: Done

File: tests/manage/rgw/test_host_node_failure.py [hunk 2]
```diff
@@ -53,25 +54,35 @@ def init_sanity(self):
         """
         self.sanity_helpers = Sanity()
 
-    def create_obc_creation(self, bucket_factory, mcg_obj, key):
+    def create_obc_creation(self, bucket_factory, obj, interface, key):
 
         # Create a bucket then read & write
-        bucket_name = bucket_factory(amount=1, interface="OC", timeout=120)[0].name
+        bucket_name = bucket_factory(amount=1, interface=interface, timeout=120)[0].name
```
[the remark this answers is not in the document]
Comment (nik-redhat) [reply tests/manage/rgw/test_host_node_failure.py]: Ack. Removed the timeout.

**Comment by nik-redhat** [comment]
I think we can merge this one, @MeridianExplorer ?

**Comment by tiffanyn108** [comment]
Re-open this PR to validate in Jenkins.

**Pull request closed without merging**